Crashed in bindStmtArgs(), when using the Server package to forward Execute statements from sysbench

I have implemented a proxy for forwarding queries using the Server package. The ServerHandler is implemented as follows:
```go
type ServerHandler struct {
	Conn  *client.Conn
}

func NewServerHandler(cfg Config) (*ServerHandler, error) {
	conn, err := client.Connect(fmt.Sprintf("%s:%d", cfg.Host, cfg.Port), cfg.User, cfg.Password, "")
	if err != nil {
		return nil, err
	}
	if err = conn.Ping(); err != nil {
		return nil, err
	}
	return &ServerHandler{
		Conn: conn,
	}, nil
}

func (h ServerHandler) UseDB(dbName string) error {
	return h.Conn.UseDB(dbName)
}

func (h ServerHandler) HandleQuery(query string) (*mysql.Result, error) {
	return h.Conn.Execute(query)
}

func (h ServerHandler) HandleFieldList(table string, fieldWildcard string) ([]*mysql.Field, error) {
	return h.Conn.FieldList(table, fieldWildcard)
}
func (h ServerHandler) HandleStmtPrepare(query string) (int, int, interface{}, error) {
	stmt, err := h.Conn.Prepare(query)
	if err != nil {
		return 0, 0, nil, err
	}
	return stmt.ParamNum(), stmt.ColumnNum(), stmt, nil
}
func (h ServerHandler) HandleStmtExecute(context interface{}, query string, args []interface{}) (*mysql.Result, error) {
	return context.(*client.Stmt).Execute(args...)
}

func (h ServerHandler) HandleStmtClose(context interface{}) error {
	return context.(*client.Stmt).Close()
}

func (h ServerHandler) HandleOtherCommand(cmd byte, data []byte) error {
	return mysql.NewError(
		mysql.ER_UNKNOWN_ERROR,
		fmt.Sprintf("command %d is not supported now", cmd),
	)
}
``` 

Unfortunately, it crashed in `(c *Conn) bindStmtArgs()` when I was testing with sysbench. It seems that when repeatedly executing the same statement with the same parameters, the EXECUTE message indicates that there is no need to re-bind the parameters, but the implementation of handleStmtExecute does not save the types and values of the parameters from the previous execution of the statement.

I tried the previously listed code with go-mysql with this PR applied: https://github.com/go-mysql-org/go-mysql/pull/983

Then the prepare ran ok:
```
$ sysbench --mysql-host=127.0.0.1 --mysql-port=4000 --mysql-user=root --mysql-db=test oltp_read_only prepare
sysbench 1.0.20 (using system LuaJIT 2.1.1713773202)

Creating table 'sbtest1'...
Inserting 10000 records into 'sbtest1'
Creating a secondary index on 'sbtest1'...
```

And then the run indeed fails as described
```
$ sysbench --mysql-host=127.0.0.1 --mysql-port=4000 --mysql-user=root --mysql-db=test oltp_read_only run --max-time=30
WARNING: --max-time is deprecated, use --time instead
sysbench 1.0.20 (using system LuaJIT 2.1.1713773202)

Running the test with following options:
Number of threads: 1
Initializing random number generator from current time


Initializing worker threads...

Threads started!

FATAL: mysql_stmt_execute() returned error 2013 (Lost connection to server during query) for query 'SELECT c FROM sbtest1 WHERE id=?'
FATAL: `thread_run' function failed: /usr/share/sysbench/oltp_common.lua:419: SQL error, errno = 2013, state = 'HY000': Lost connection to server during query
```

With the server reporting this:
```
$ go run main.go 
panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
github.com/go-mysql-org/go-mysql/server.(*Conn).bindStmtArgs(0xc00002fd18?, 0xc0004a6000, {0xc0004881d2, 0x1, 0x0?}, {0x0, 0x0, 0xc00002fd98?}, {0x0, 0x0, ...})
	/home/dvaneeden/dev/go-mysql/server/stmt.go:192 +0xd10
github.com/go-mysql-org/go-mysql/server.(*Conn).handleStmtExecute(0xc0000bc2c0, {0xc0004881c9, 0x13, 0x17})
	/home/dvaneeden/dev/go-mysql/server/stmt.go:160 +0x1e9
github.com/go-mysql-org/go-mysql/server.(*Conn).dispatch(0xc0000bc2c0, {0xc0004881c8?, 0x14, 0x18})
	/home/dvaneeden/dev/go-mysql/server/command.go:121 +0x43b
github.com/go-mysql-org/go-mysql/server.(*Conn).HandleCommand(0xc0000bc2c0)
	/home/dvaneeden/dev/go-mysql/server/command.go:59 +0xa6
main.main()
	/tmp/issue915/main.go:90 +0x105
exit status 2
```

--- a/server/stmt.go
+++ b/server/stmt.go
@@ -176,7 +176,10 @@ func (c *Conn) handleStmtExecute(data []byte) (*mysql.Result, error) {
 func (c *Conn) bindStmtArgs(s *Stmt, nullBitmap, paramTypes, paramValues []byte) error {
 	args := s.Args
 
-	// Every param should have a 1-byte type and a 1-byte flag
+	// Every param should have a type-and-flag of 2 bytes
+	// 0xfe80 == Type 0xfe and Flag 0x80
+	// The flag only has one bit and that indicates if it is unsigned or not.
+	// Types are 1 byte, but might grow into the 7 unused bits in the future.
 	if len(paramTypes)/2 != s.Params {
 		return mysql.ErrMalformPacket
 	}

--- a/server/stmt.go
+++ b/server/stmt.go
@@ -155,10 +155,10 @@ func (c *Conn) handleStmtExecute(data []byte) (*mysql.Result, error) {
 			pos += paramNum << 1
 
 			paramValues = data[pos:]
-		}
 
-		if err := c.bindStmtArgs(s, nullBitmaps, paramTypes, paramValues); err != nil {
-			return nil, errors.Trace(err)
+			if err := c.bindStmtArgs(s, nullBitmaps, paramTypes, paramValues); err != nil {
+				return nil, errors.Trace(err)
+			}
 		}
 	}
 
@@ -176,6 +176,11 @@ func (c *Conn) handleStmtExecute(data []byte) (*mysql.Result, error) {
 func (c *Conn) bindStmtArgs(s *Stmt, nullBitmap, paramTypes, paramValues []byte) error {
 	args := s.Args
 
+	// Every param should have a 1-byte type and a 1-byte flag
+	if len(paramTypes)/2 != s.Params {
+		return mysql.ErrMalformPacket
+	}
+
 	pos := 0
 
 	var v []byte
@@ -190,7 +195,7 @@ func (c *Conn) bindStmtArgs(s *Stmt, nullBitmap, paramTypes, paramValues []byte)
 		}
 
 		tp := paramTypes[i<<1]
-		isUnsigned := (paramTypes[(i<<1)+1] & 0x80) > 0
+		isUnsigned := (paramTypes[(i<<1)+1] & mysql.PARAM_UNSIGNED) > 0
 
 		switch tp {
 		case mysql.MYSQL_TYPE_NULL:

